From: =?utf-8?q?Jonas_Dre=C3=9Fler?= <verdre@v0yd.nl>
Date: Sun, 10 Apr 2022 22:46:12 +0200
Subject: clutter/actor-meta: Notify actor property on changes

Seems like this was forgotten when writing the class. While at it, add
EXPLICIT_NOTIFY to the property flags.

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/2377>
Origin: upstream, 42.1, commit:6c8e8fbba45ce881871f478cd65dc8aafe2aef09
---
 clutter/clutter/clutter-actor-meta.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/clutter/clutter/clutter-actor-meta.c b/clutter/clutter/clutter-actor-meta.c
index 02bbc01..8b564ea 100644
--- a/clutter/clutter/clutter-actor-meta.c
+++ b/clutter/clutter/clutter-actor-meta.c
@@ -109,6 +109,8 @@ clutter_actor_meta_real_set_actor (ClutterActorMeta *meta,
     priv->destroy_id = g_signal_connect (priv->actor, "destroy",
                                          G_CALLBACK (on_actor_destroy),
                                          meta);
+
+  g_object_notify_by_pspec (G_OBJECT (meta), obj_props[PROP_ACTOR]);
 }
 
 static void
@@ -213,7 +215,8 @@ clutter_actor_meta_class_init (ClutterActorMetaClass *klass)
                          P_("Actor"),
                          P_("The actor attached to the meta"),
                          CLUTTER_TYPE_ACTOR,
-                         CLUTTER_PARAM_READABLE);
+                         CLUTTER_PARAM_READABLE |
+                         G_PARAM_EXPLICIT_NOTIFY);
 
   /**
    * ClutterActorMeta:name:
